gwr-spotter/src/app/tests.rs

```diff
@@ -0,0 +1,410 @@
+// Copyright (c) 2025 Graphcore Ltd. All rights reserved.
+
+#[cfg(test)]
+mod tests {
+    // Note that all tests need to be run serially as spotter uses a global state
+    // shared between threads.
```

Comment:
Should the `SHARED_STATE` be reset between test cases, in `create_test_app()` for example?